executor/src/witgen/global_constraints.rs

```diff
@@ -268,8 +266,67 @@ fn propagate_constraints<T: FieldElement>(
     (known_constraints, remove)
 }
 
+/// Get simple expression roots for a given expression.
```

Comment:
Please tell me if the following is correct:

Tries to return the set of roots of a given expression in one variable.

If `Some(vec![r1, r2, ... ,rn])` is returned, then the expression is equivalent to `(x - r1)*(x - r2) * ... * (x - rn)`.

Please also change this function to return a tuple of the variable an the vector of roots. Or do you think this return type is better suited in some way?

Reply:
Oh, the above is actually not true, it should be equivalent to a nonzero multiple of the product.

Reply:
> If Some(vec![r1, r2, ... ,rn]) is returned, then the expression is equivalent to (x - r1)*(x - r2) * ... * (x - rn).

This one is correct. If I understood correctly, you need to support (x-r1)*(x-r2)*...*(y-r'1)*(y-r'2)*...
and return [(x, [r1,r2,...]),(y,[r'1, r'2,...])]

Reply:
No, we cannot support that, for the pattern to match, there can only even be one variable.